packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1748,17 +1756,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
     _lastKnownRemoteTextEditingValue = value;
 
-    if (value == _value) {
-      // This is possible, for example, when the numeric keyboard is input,
-      // the engine will notify twice for the same value.
-      // Track at https://github.com/flutter/flutter/issues/65811
-      return;
+    final bool shouldShowCaret = widget.readOnly
+      ? _value.selection != value.selection
+      : _value != value;
+    if (shouldShowCaret) {
+      _scheduleShowCaretOnScreen();
```

Comment:
I didnt get the logic, the _scheduleShowCaretOnScreen will be called in line 1770 anyway?

Reply:
Ah I forgot to remove the existing `_sheduleShowCaretOnScreen` when copied the other one over.